drivers/ieee802154/ieee802154_efr32.c

```diff
@@ -0,0 +1,558 @@
+                    dev, NULL, NULL, K_PRIO_COOP(2), 0, 0);
+
+    LOG_DBG("Init done!");
+    return 0;
+}
+
+static void efr32_received()
+{
+    // RAIL_RxPacketHandle_t packetHandle = RAIL_RX_PACKET_HANDLE_INVALID;
+    // RAIL_RxPacketInfo_t packetInfo;
+    // RAIL_RxPacketDetails_t packetDetails;
+
+    // RAIL_Status_t status;
+    // uint16_t length;
+
+    // packetHandle = RAIL_GetRxPacketInfo(efr32_data.rail_handle, RAIL_RX_PACKET_HANDLE_OLDEST, &packetInfo);
+}
+
+static void efr32_rail_cb(RAIL_Handle_t rail_handle, RAIL_Events_t events)
+{
+    /*
+                                   RAIL_EVENT_RX_ACK_TIMEOUT |                      //
+                                   RAIL_EVENT_TX_PACKET_SENT |                  //
+                                   RAIL_EVENT_RX_PACKET_RECEIVED |              //
+                                   RAIL_EVENT_TX_CHANNEL_BUSY |                 //
+                                   RAIL_EVENT_TX_ABORTED |                      //
+                                   RAIL_EVENT_TX_BLOCKED |                      //
+                                   RAIL_EVENT_TX_UNDERFLOW |                    //
+                                   RAIL_EVENT_IEEE802154_DATA_REQUEST_COMMAND | //
+                                    RAIL_EVENT_CAL_NEEDED                        //*/
+    LOG_DBG("Processing events %d", sizeof(events));
+
+    size_t index = 0;
+    do
+    {
+        switch (index)
+        {
+        case RAIL_EVENT_RX_ACK_TIMEOUT:
+            LOG_DBG("RX AutoAck occurred");
+            break;
+        case RAIL_EVENT_TX_ABORTED:
+            LOG_DBG("TX was aborted");
+            break;
+        case RAIL_EVENT_TX_BLOCKED:
+            LOG_DBG("TX is blocked");
+            break;
+        case RAIL_EVENT_TX_UNDERFLOW:
+            LOG_DBG("TX is underflow");
+            break;
+        case RAIL_EVENT_TX_PACKET_SENT:
+            LOG_DBG("TX packet sent");
+            break;
+        case RAIL_EVENT_RX_PACKET_RECEIVED:
+            LOG_DBG("Received packet frame");
+            efr32_received();
+            break;
+        }
+        events = events >> 1;
+        index += 1;
+    } while (events != 0);
+}
+
+static struct ieee802154_radio_api efr32_radio_api = {
+    .iface_api.init = efr32_iface_init,
+    .iface_api.send = ieee802154_radio_send,
+
+    .get_capabilities = efr32_get_capabilities,
+    .cca = efr32_cca,
+    .set_channel = efr32_set_channel,
+    .filter = efr32_filter,
+    .set_txpower = efr32_set_txpower,
+    .start = efr32_start,
+    .stop = efr32_stop,
+    .tx = efr32_tx,
+};
+
+#if defined(CONFIG_NET_L2_IEEE802154)
+
+#define L2 IEEE802154_L2
+#define L2_CTX_TYPE NET_L2_GET_CTX_TYPE(IEEE802154_L2)
+#define MTU 125
+
+#elif defined(CONFIG_NET_L2_OPENTHREAD)
+#define L2 OPENTHREAD_L2
+#define L2_CTX_TYPE NET_L2_GET_CTX_TYPE(OPENTHREAD_L2)
+#define MTU 1280
+
+#endif
+
+NET_DEVICE_INIT(
+    efr32,                             /* Device Name */
```

Comment:
the comments are superfluous imo, the NET_DEVICE_INIT macro is documented already in net_if.h